mesa/space.py

```diff
@@ -256,29 +254,37 @@ def get_neighborhood(
         if neighborhood is not None:
             return neighborhood
 
-        coordinates: set[Coordinate] = set()
-
         x, y = pos
-        for dy in range(-radius, radius + 1):
-            for dx in range(-radius, radius + 1):
-                # Skip coordinates that are outside manhattan distance
-                if not moore and abs(dx) + abs(dy) > radius:
-                    continue
 
-                coord = (x + dx, y + dy)
+        ls_ranges_x = self.inbound_ranges(x, self.width, radius)
+        ls_ranges_y = self.inbound_ranges(y, self.height, radius)
 
-                if self.out_of_bounds(coord):
-                    # Skip if not a torus and new coords out of bounds.
-                    if not self.torus:
-                        continue
-                    coord = self.torus_adj(coord)
+        neighborhood = []
 
-                coordinates.add(coord)
+        for range_x, range_y in itertools.product(ls_ranges_x, ls_ranges_y):
 
-        if not include_center:
-            coordinates.discard(pos)
+            for nx in range_x:
+                for ny in range_y:
+                    if nx == x and ny == y and not include_center:
+                        continue
+                    if not moore:
+                        dx, dy = abs(nx - x), abs(ny - y)
+                        if self.torus:
+                            # In the torus case the deviation from the given point
+                            # can derive from traversing the grid from the outside
+                            dx_out = self.width - dx
+                            dy_out = self.height - dy
+                            if dx_out < dx:
+                                dx = dx_out
```

Comment:
You can do the whole thing in 1 line with `dx = min(dx, self.width - dx)`

Reply:
mmh It's much slower this way unfortunately

Reply:
do you think going that way must be dome the same?

Reply:
Yes, `:=` sounds good, though I'm rather surprised that `min(...)` is significantly slower. By how much? It's also used in https://github.com/projectmesa/mesa/blob/952857c7accdd696cd791474e9f44e4f2a90988a/mesa/space.py#L936. If we implement our own `min`, which does the same as using the `if` statement in this line, would it be faster?

Reply:
```python
def get_min(a, b):
  return a if a < b else b
```

Reply:
```ipython
In [13]: np.mean(
    ...:     timeit.repeat(
    ...:         "f(4,5)", setup="f = lambda a, b: a if a < b else b", repeat=20, number=100
    ...:     )
    ...: )
Out[13]: 4.7881399950711057e-05

In [14]: np.mean(timeit.repeat("min(4,5)", repeat=20, number=100))
Out[14]: 7.098469868651591e-05
```

Reply:
Yet three 2.5 times slower than an inline expression:
min(timeit.repeat("a if a < b else b", setup="a=4; b=5", repeat=20, number=100000)) -> 0.0037641999078
min(timeit.repeat("f(4,5)", setup="f = lambda a, b: a if a < b else b", repeat=20, number=100000)) -> 0.0103960999986

Reply:
Applied the `:=` operator